src/pages/Content/index.js

```diff
@@ -31,7 +31,22 @@ function click(selector) {
 }
 // The function plays a recording when needed
 function playRecording(recordedEvents) {
-  recordedEvents.forEach(element => {
-    click(element);
-  });
+  let intervalId = setInterval(() => {
+    if (document.readyState === 'complete') {
+      for (const element of recordedEvents) {
+        const elem = document.querySelector(element) || null;
+        let status = false;
+        if (elem) {
+          elem.addEventListener('click', () => { status = true }, true); // the callback will be invoked only once
```

Comment:
Why are you adding an event listener?
You can just add a `throw new Exception(...)` in `click`, and `try`/`catch` it here.

Reply:
Implemented try/catch as advised.